### PR TITLE
feature: BOTI cancelling

### LINKED ISSUE #294: Add cancelling mechanism to BOTI

### PR DESCRIPTION
**Pull request by IskandarGhobril**
This PR introduces a cancelling mechanism for the BOTI battle flow. Users now have the option to cancel their request if the opponent hasn't accepted it yet. Additionally:  
- Navigating away from the "Battle Request Sent" screen will automatically cancel the request.  
- Navigation between certain screens has been restricted to ensure both users remain at the same stage of the battle.  

cancel

#### Changes:
- **`BattleViewModel.kt`**:  
  - Added a `cancelBattle()` function to update the battle status to "Cancelled."  

- **`BattleRequestSentScreen.kt`**:  
  - Added a **Cancel button** to allow users to cancel the battle request.  
  - Ensured that any exit (e.g., navigation) from this screen cancels the request if it hasn’t been accepted.  

- **`BattleChatScreen.kt`**:  
  - Removed the back button to prevent exiting from the chat.  

- **`EvaluationScreen.kt`**:  
  - Refactored the action button to allow customization of button text and background colors.  

- **`ChatScreen.kt`**:  
  - Made the display of the back button optional to enable its removal in the "Battle Chat" screen.  

- **`BattleChatScreenTest.kt`**:  
  - Updated tests to validate that the back button is no longer displayed.  

This PR closes #294

### CODE REVIEW

File: app/src/main/java/com/github/se/orator/ui/battle/BattleRequestSentScreen.kt [hunk 1]
```diff
@@ -85,11 +119,27 @@ fun BattleRequestSentScreen(
...
+
+              // Cancel Button
+              ActionButton(
+                  "Cancel Battle",
+                  onClick = {
+                    battleViewModel.cancelBattle(battleId)
+                    navigationActions.navigateTo(TopLevelDestinations.HOME)
+                    Toast.makeText(context, "Battle has been cancelled!", Toast.LENGTH_SHORT).show()
+                  },
+                  textColor = Color.White,
```
Comment (ymarc11) [inline app/src/main/java/com/github/se/orator/ui/battle/BattleRequestSentScreen.kt:141]: You have hard coded colors here. Consider putting them in Color.kt.

**Review by ymarc11 — CHANGES_REQUESTED** [review]
This PR introduces a good cancellation mechanism for the BOTI. It enhances user experience by providing control over pending battle requests while preventing unwanted navigation to inconsistent battle stages. The CI passes, 0 code duplication and the test coverage is over 80%, well done.

Cancel property:

- The cancelBattle() function is cleanly implemented and improves status management.
- The "Cancel" button addition enhances user flexibility.
- The use of DisposableEffect for lifecycle handling is excellent for auto cancellations.

Back button:
- The update of the back button to cancel the battle once clicked is good.

Tests:
- Test updates confirm that the back button is hidden as expected.

Suggestions:
- You could put the colors used in the "Cancel" action button in Color.kt

This PR improves the BOTI flow with thoughtful user interaction design (the ability to cancel) and clean navigation management. Overall, the changes are well-structured, and the functionality aligns with the feature goals. Ready to be merged, just fix the hard coded colors for better theme management.

File: app/src/main/java/com/github/se/orator/ui/battle/BattleRequestSentScreen.kt (hunk 1 above)
Comment (IskandarGhobril) [reply app/src/main/java/com/github/se/orator/ui/battle/BattleRequestSentScreen.kt:141]: That is right. Good catch!

**Comment by IskandarGhobril** [comment]
Hey @ymarc11 I implemented the requested changes. Can you please approve the PR? Thanks

**Review by ymarc11 — APPROVED** [review]

**Pull request merged**